src/commands/dev/dev.mjs

```diff
@@ -55,16 +55,29 @@ import { createDevExecCommand } from './dev-exec.mjs'
 const netlifyBuildPromise = import('@netlify/build')
 
 const startStaticServer = async ({ settings }) => {
-  const server = new StaticServer({
-    rootPath: settings.dist,
-    name: 'netlify-dev',
-    port: settings.frameworkPort,
-    templates: {
-      notFound: path.join(settings.dist, '404.html'),
-    },
+  const server = Fastify()
+  const rootPath = path.resolve(settings.dist)
+  server.register(fastifyStatic, {
+    root: rootPath,
+    etag: false,
+    acceptRanges: false,
+    lastModified: false,
   })
 
-  await promisify(server.start.bind(server))()
+  server.setNotFoundHandler((_req, res) => {
+    res.code(404).sendFile('404.html', rootPath)
+  })
+
+  server.addHook('onRequest', (req, reply, done) => {
+    reply.header('X-Powered-by', 'netlify-dev')
```

Comment:
The `name: 'netlify-dev'` configuration in `static-server` used to set the `X-Powered-by` header.